Dockerfile

```diff
@@ -6,13 +6,30 @@ WORKDIR /home/getml
 
 COPY --chown=getml:getml --chmod=0777 ./requirements.txt /home/getml/requirements.txt
 
-ENV PATH="$PATH:/home/getml/.local/bin"
+ENV PATH="/home/getml/.local/bin:$PATH"
+
 RUN python3.11 \
     -mpip install \
     -r /home/getml/requirements.txt
 
-RUN mkdir /home/getml/.getML /home/getml/.getML/logs /home/getml/.getML/projects /home/getml/demo
-RUN curl https://storage.googleapis.com/static.getml.com/download/1.4.0/getml-1.4.0-x64-linux.tar.gz | tar -C /home/getml/.getML -xvzf -
+ARG TARGETOS
+ARG TARGETARCH
+ARG GETML_VERSION_NUMBER
+
+RUN mkdir /home/getml/.getML
+
+RUN <<EOT
+    #!/bin/bash
+    if [ "${TARGETARCH}" = "amd64" ]; then
+        export GETML_ARCH="x64"
+    else
+        export GETML_ARCH="${TARGETARCH}"
+    fi;
+    curl "https://storage.googleapis.com/static.getml.com/download/${GETML_VERSION_NUMBER}/getml-${GETML_VERSION_NUMBER}-"`echo \${GETML_ARCH}`"-${TARGETOS}.tar.gz" | 
```

Comment:
Does the hack require the use of backticks? I generally prefer using `$()` for command substitutions because there are some weird edge cases and limitations when using backticks.

http://mywiki.wooledge.org/BashFAQ/082

Reply:
Thinking about this, the nicest way would be to use a docker-stage ternary for this. This way we can save us the inline script:

```Dockerfile
FROM python:3.11.8 as base
RUN useradd getml
USER getml
WORKDIR /home/getml

COPY --chown=getml:getml --chmod=0777 ./requirements.txt /home/getml/requirements.txt

ENV PATH="/home/getml/.local/bin:$PATH"

RUN python3.11 \
    -mpip install \
    -r /home/getml/requirements.txt

ARG TARGETOS
ARG TARGETARCH
ARG GETML_VERSION_NUMBER

FROM base as base-amd64
ARG GETML_ARG=x64

FROM base as base-arm64
ARG GETML_ARG=arm64

FROM base-${TARGETARCH}
ARG BUCKET="https://storage.googleapis.com/static.getml.com/download"
ARG TARBALL="getml-${GETML_VERSION_NUMBER}-${GETML_ARG}-${TARGETOS}.tar.gz"
ARG DOWNLOAD_URL="${BUCKET}/${GETML_VERSION_NUMBER}/${TARBALL}"

RUN mkdir /home/getml/.getML

RUN curl -L ${DOWNLOAD_URL} | tar -C /home/getml/.getML -xvzf -

COPY --chown=getml:getml . /home/getml/demo/

EXPOSE 1709 8888
CMD [ "/home/getml/.local/bin/jupyter", "lab", "--ip='*'", "--port=8888", "--notebook-dir='/home/getml/demo'" ]
```